frontend/src/routes/Video.tsx

```diff
@@ -641,26 +644,38 @@ const ShareButton: React.FC<{ event: SyncedEvent }> = ({ event }) => {
 
     const inner = match(state, {
         "closed": () => null,
-        "main": () => <>
-            <CopyableInput
-                label={t("manage.my-videos.details.copy-direct-link-to-clipboard")}
-                css={{ fontSize: 14, width: 400 }}
-                // TODO
-                value={window.location.href}
-            />
-            <ShowQRCodeButton target={window.location.href} label={state} />
-        </>,
+        "main": () => {
+            let url = window.location.href.replace(regEx, "");
+            if (timestamp) {
+                url = url + `?t=${timestamp}`;
+            }
+            return <>
+                <div>
+                    <CopyableInput
+                        label={t("manage.my-videos.details.copy-direct-link-to-clipboard")}
+                        css={{ fontSize: 14, width: 400 }}
+                        // TODO
+                        value={url}
+                    />
+                    <TimePicker {...{ setTimestamp }} />
+                </div>
+                <ShowQRCodeButton target={window.location.href} label={state} />
+            </>;
+        },
         "embed": () => {
             const ar = event.syncedData == null
                 ? [16, 9]
                 : getPlayerAspectRatio(event.syncedData.tracks);
 
-            const target = new URL(location.href);
-            target.pathname = `/~embed/!v/${event.id.slice(2)}`;
+            let url = new URL(location.href.replace(regEx, ""));
+            if (timestamp) {
+                url = new URL(url + `?t=${timestamp}`);
+            }
```

Comment:
This could use `url.searchParams.set(...)` (`searchParams` is readOnly otherwise), but I don't think that works with a ternary expression. Your suggested solution works for the embed route, but since the url in `main` is just a string, it would first need to be built as `URL` and then be cast as `string` again when given to the `CopyableInput`, so I think it's better to just keep it a `string`.